README.md

```diff
@@ -1,102 +1,262 @@
 
-  # Returns user object, if the login attempt was valid.
-  sp.assert(idp, request_body, cb)
+  // Initialize options object
+  var sp_options = {
+    entity_id: "https://sp.example.com/metadata.xml",
+    private_key: fs.readFileSync("key-file.pem").toString(),
+    certificate: fs.readFileSync("cert-file.crt").toString(),
+    assert_endpoint: "https://sp.example.com/assert",
+    force_authn: true,
+    auth_context: { comparison: "exact", class_refs: ["urn:oasis:names:tc:SAML:1.0:am:password"] },
+    nameid_format: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+    sign_get_request: false,
+    allow_unencrypted_assertion: true
+  }
 
-  # -- OPTIONAL --
-  # Returns a redirect URL, at which a user is logged out.
-  sp.create_logout_url(idp, cb)
+  // Call service provider constructor with options
+  var sp = saml2.ServiceProvider(sp_options);
+
+  // Example use of service provider.
+  // Call metadata to get XML metatadata used in configuration.
+  var metadata = sp.create_metadata();
 
-  # Returns XML containing service-provider parameters.
-  # For use during initial SAML configuration
-  sp.create_metadata(idp, cb)
 ```
 
-## Helper Methods
+#### Service provider function definitions
 
-We will break each of the `service_provider` methods into minimal, testable methods.
+<a name="create_login_request_url" />
+##### create_login_request_url(IdP, options, cb)
+Get a URL to initiate a login.
 
-```coffee
-  ... TODO ...
-  parse_xml
-  parse_assert
-  createAuthRequest
-```
+Takes the following arguments:
+- `IdP` - [IdP](#IdentityProvider)
+- `options`
+  - `relay_state` - SAML relay state.
+  - `auth_context` - Specifies `AuthnContextClassRef`.  This can also be configured on the [SP](#ServiceProvider).
+  - `nameid_format` - Format for Name ID.  This can also be configured on a per-method basis.
+  - `force_authn` - Boolean that allows forcing re-authentication of users even if the user has a SSO session with the IdP.  This can also be configured on the [IdP](#IdentityProvider) or [SP](#ServiceProvider).
+  - `sign_get_request` - Boolean, if true signs the request.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+- `cb(error, url, request_id)` - Callback called with the login URL and ID of the request.
+
+
+<a name="redirect_assert" />
+##### redirect_assert(IdP, options, cb)
+Gets a user object if the login attempt is valid, inflates the response before parsing it.
+
+Takes the following arguments:
+- `IdP` - [IdP](#IdentityProvider)
+- `options`
+  - `request_body` - Object with a `SAMLResponse` or `SAMLRequest`.
+  - `allow_unencrypted_assertion` - Boolean, when set to true allows unencrypted assertions.  This can also be configured on the [IdP](#IdentityProvider) or [SP](#ServiceProvider).
+- `cb(error, response)` - Callback called with the request response
+
+
+<a name="post_assert" />
+##### post_assert(IdP, options, cb)
+Gets a user object if the login attempt is valid.
```

Comment:
4267b02c517fa705dc51a668a973aa4b2f4e5eb2
I made it more clear that it's used specifically for post binding, hopefully this is enough information for someone trying to use saml.  http://en.wikipedia.org/wiki/SAML_2.0#HTTP_POST_Binding

Does that seem reasonable?  We could also point to documents that have more in depth descriptions?